test/python/test_certificate_authority_api.py

```diff
@@ -0,0 +1,297 @@
+            )
+
+        if key_password:
+            self.secrets_api.create_variable(
+                self.account,
+                'variable',
+                f'conjur/{self.CA_SERVICE_ID}/ca/private-key-password',
+                body=key_password
+            )
+
+    def tearDown(self):
+        super().load_default_policy()
+
+    def test_sign_201_json(self):
+        """Test case for 201 JSON response when requesting a signed certificate
+        The default response to a successful request is a JSON response body
+        with a `certificate` field including the signed certificate
+        """
+        response, status, _ = self.api.sign_with_http_info(
+            self.account,
+            self.CA_SERVICE_ID,
+            self.csr,
+            'P1D'
+        )
+
+        self.assertEqual(status, 201)
+        self.assertIsInstance(response, openapi_client.models.certificate_json.CertificateJson)
+
+        self.assertEqual(response.certificate[:27], '-----BEGIN CERTIFICATE-----')
+
+    @unittest.expectedFailure
+    def test_sign_201_pem(self):
+        """Test case for 201 PEM response when requesting a signed certificate
+        When the `Accept` header is set to `application/x-pem-file`, the request body
+        is a PEM encoded certificate
+
+        This test case fails due to an issue in the OpenAPI Generator
+        with passing an `Accept` header. The generator overwrites any assignment to `Accept`.
+        """
+        response, status, _ = self.api.sign_with_http_info(
+            self.account,
+            self.CA_SERVICE_ID,
+            self.csr,
+            'P1D',
+            accept='application/x-pem-file'
+        )
+
+        self.assertEqual(status, 201)
+        self.assertIsInstance(response, str)
+
+    def test_sign_with_encrypted_key_201(self):
+        """Test case for 201 response when requesting a signed certificate
+        Uses an encrypted intermediate key and password to sign CSR
+        """
+        key_password = read_file(KEY_PASSWORD_PATH)
+        encrypted_key = read_file(ENCRYPTED_KEY_PATH)
+
+        self.configure_ca(None, encrypted_key, key_password)
+
+        response, status, _ = self.api.sign_with_http_info(
+            self.account,
+            self.CA_SERVICE_ID,
+            self.csr,
+            'P1D'
+        )
+
+        self.assertEqual(status, 201)
+        self.assertIsInstance(response, openapi_client.models.certificate_json.CertificateJson)
+
+        self.assertEqual(response.certificate[:27], '-----BEGIN CERTIFICATE-----')
+
+    def test_sign_with_encrypted_key_500(self):
+        """Test case for 500 response when requesting a signed certificate
+        500 status repsonses can result from a misconfigured CA service
+        In this test, the Conjur variable for the encrypted key's password is incorrect
+        """
+        encrypted_key = read_file(ENCRYPTED_KEY_PATH)
+
+        self.configure_ca(None, encrypted_key, 'wrong_pass')
```

Comment:
When calling a method with kwargs like this it is better to specify the kwarg names. You can also omit the `None` parameter here because it is already the default. e.g. `self.configure_ca(private_key=encrypted_key, key_password='wrong_pass')`